client/packages/programs/src/JsonForms/common/components/Array.tsx

```diff
@@ -254,7 +297,114 @@ const ArrayComponent = (props: ArrayControlCustomProps) => {
     label,
     rootSchema,
     renderers,
+    config,
   } = props;
+  const { errors: zErrors, options } = useZodOptionsValidation(
+    Options,
+    props.uischema.options
+  );
+
+  const defaultExpanded = props.uischema?.options?.['defaultExpanded'] ?? false;
+
+  const [expandedItems, setExpandedItems] = useState<boolean[]>(
+    new Array(inputData?.length ?? 0).fill(defaultExpanded)
+  );
+
+  const isNotesArray = options?.detail?.elements[0]?.type === 'Note';
```

Comment:
Some behaviour is Note-specific, so we identify here whether we're dealing with notes or not. As mentioned, I'll pull some of the note-specific code out into another file once we're done here to make this module a little less bloated.